dlt/dlt.py

```diff
@@ -822,6 +825,10 @@ def __iter__(self):  # pylint: disable=too-many-branches
                     if not index % 100000:
                         self._log_message_progress()
                     yield msg
+                else:
+                    if self.live_run and self.iterate_unblock_mode:
+                        logger.debug("Yield None when no new message arrives [at %s]", self.file_position)
+                        yield None
```

Comment:
where i added "yield None" are not the places of raising error, so if there is error, then it will still be thrown out.
The added "yield None" are meant to unblock  "for message in cDLTFile" in the main loop (secondary loop), and the main loop is used in the for-loop (first loop) inside DLTFileSpinner.run(). If secondary loop is blocked at the moment of no more message from dlt file, then first loop is also blocked and could not handle further actions.

Please review DLTFileSpinner. If you reject to do that, then you cannot get the whole picture.

DLTFileSpinner is another use case of DLTMessageHandler, for such absolute truth, i knew that from the very beginning. And that is also why Wenshi does not like my way of copy-and-paste the majority part of DLTMessageHandler code, in order to achieve "not touching the branch of DLTConnector". I dont mean the ugly state is the end state, I'm aware of what I'm doing, what I have in mind is to split the big and complicated task into two smaller steps:
1. simple append a new functionality for IOC to mtee, without touching DLTConnector from DLTClient
2. code refactory and bridge DLTFileSpinner and DLTMessageHandler together. At this step, only python-dlt needs to be worked on, i dont even need to touch mtee_core, mtee_gen22, and idcevo. Because the used API will stay the same there in the uplevel.

what I want to achieve with such split: in case somebody complains DLTConnector does not work after this big change, then I dont need to revoke all 2 steps, right? because step 1 does not touch DLTConnector with DLTClient at all. I just need to focus on the change made in step2, and the worst case is to revoke step 2 from python-dlt without change mtee_core, mtee_gen22, idcevo, because the added IOC functionality introduced from step 1 stays there.